homeassistant/components/zabbix/__init__.py

```diff
@@ -46,13 +90,208 @@ def setup(hass, config):
     username = conf.get(CONF_USERNAME)
     password = conf.get(CONF_PASSWORD)
 
-    zapi = ZabbixAPI(url)
+    publish_states_host = conf.get(CONF_PUBLISH_STATES_HOST)
```

Comment:
Please use `dict[key]` when we know the key is in the dict.
```suggestion
    publish_states_host = conf[CONF_PUBLISH_STATES_HOST]
```

Reply:
CONF_PUBLISH_STATES_HOST is optional

Reply:
There's a default in the config shema.

Reply:
Bump

Reply:
Ah I see. So the default will be put into Dict if it is not specified in the config. Will change to square brackets then.

Reply:
Although I think removing the default is the more appropriate solution.